### PR TITLE
New table: ntfs_journal_events

### PR DESCRIPTION
**Pull request by woodruffw**
Example config (monitors the `Downloads` directory, as well as all files within it at configure-time):

```json
{
  "file_paths": {
    "downloads": [
      "C:\\Users\\tob\\Downloads",
      "C:\\Users\\tob\\Downloads\\*"
    ]
  }
}
```

### CODE REVIEW

File: osquery/tables/events/windows/ntfs_journal_events.cpp [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+  }
+
+  if (!emitted_row_list.empty()) {
+    addBatch(emitted_row_list);
+  }
+
+  return Status(0);
+}
+
+// TODO(alessandro): Write a test for this
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply osquery/tables/events/windows/ntfs_journal_events.cpp]: Yup! I probably also have to re-integrate the existing tests into the Buck build, so I'll do this as part of that.

File: osquery/events/windows/ntfs_event_publisher.cpp [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+      rename_path_mapper.erase(it);
+    }
+    }
+
+    if (skip_record) {
+      continue;
+    }
+
+    // Generate the new event
+    NTFSEventRecord event = {};
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply osquery/events/windows/ntfs_event_publisher.cpp]: Makes sense to me!

File: osquery/events/windows/ntfs_event_publisher.cpp [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+
+NTFSEventPublisherConfiguration NTFSEventPublisher::readConfiguration() {
+  auto config_parser = Config::getParser("file_paths");
+  const auto& json = config_parser->getData().doc();
+
+  NTFSEventPublisherConfiguration configuration = {};
+
+  // We are not going to expand the paths, as we just need to get the
+  // drive letter in order to restart the reader services
+  Config::get().files(
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply osquery/events/windows/ntfs_event_publisher.cpp]: osquery's configuration API is a bit of a mystery to me (I didn't write those original lines). FWICT, `getParser()` does indeed set some global (or function-static) state that `Config::get()` then relies on. I'll do some experimentation.

[the remark this answers is not in the document]
Comment (woodruffw) [reply osquery/events/windows/ntfs_event_publisher.cpp]: Did some tests, and removing the lines had no effect. So 🤷‍♀️ on them being there in the first place.

[the remark this answers is not in the document]
Comment (GarretReece) [reply osquery/events/windows/ntfs_event_publisher.cpp]: Probably vestigial from when I first wrote it and cargo culted it in

File: osquery/events/windows/usn_journal_reader.cpp [hunk 4]
```diff
@@ -0,0 +1,879 @@
...
+  // We don't care about range tracking records
+  if (record->MajorVersion == 4U) {
+    return Status(0);
+  }
+
+  if (record->MinorVersion != 0U) {
+    LOG(WARNING) << "Unexpected minor version value";
+  }
+
+  USNJournalEventRecord base_event_record = {};
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply osquery/events/windows/usn_journal_reader.cpp]: Fiddled with this a bit, but there's enough default constructor use elsewhere and reliance on `Status` returns to make it slightly annoying. Perhaps with a separate initialization method, but I'll put this on the shelf until the actual upstream PR.

**Comment by woodruffw** [comment]
https://github.com/facebook/osquery/pull/5371

**Pull request closed without merging**